src/http.rs

```diff
@@ -123,6 +124,23 @@ where
                 })
                 .context(CallRequestSnafu)?;
 
+            let (parts, body) = response.into_parts();
+            let mut data_buf: Option<Vec<u8>> = Some(Vec::new());
+
+            let body = body
+                .map_data(move |data| {
+                    if let Some(buf) = &mut data_buf {
+                        buf.extend_from_slice(&data);
```

Comment:
Pursuant to @binarylogic's comment, we should also limit the number of bytes added to the slice here. That is, if the incoming data is 1MB and we only want the first 100 bytes, there's no point copying it all only to truncate. Limiting the size of `buf` would also mean we don't need to truncate below.